src/webgpu/api/validation/capability_checks/limits/maxTextureDimension2D.spec.ts

```diff
@@ -0,0 +1,86 @@
+import { getGPU } from '../../../../../common/util/navigator_gpu.js';
+import { kAllCanvasTypes, createCanvas } from '../../../../util/create_elements.js';
+
+import { kLimitBaseParams, makeLimitTestGroup } from './limit_utils.js';
+
+const limit = 'maxTextureDimension2D';
+export const { g, description } = makeLimitTestGroup(limit);
+
+g.test('createTexture,at_over')
+  .desc(`Test using at and over ${limit} limit`)
+  .params(kLimitBaseParams)
+  .fn(async t => {
+    const { limitTest, testValueName } = t.params;
+    await t.testDeviceWithRequestedLimits(
+      limitTest,
+      testValueName,
+      async ({ device, shouldError, testValue, actualLimit }) => {
+        for (let dimensionIndex = 0; dimensionIndex < 2; ++dimensionIndex) {
+          const size = [1, 1, 1];
+          size[dimensionIndex] = testValue;
+
+          await t.testForValidationErrorWithPossibleOutOfMemoryError(
+            () => {
+              const texture = device.createTexture({
+                size,
+                format: 'rgba8unorm',
+                usage: GPUTextureUsage.TEXTURE_BINDING,
+              });
+
+              // MAINTENANCE_TODO: Remove this 'if' once the bug in chrome is fixed
+              // This 'if' is only here because of a bug in Chrome
+              // that generates an error calling destroy on an invalid texture.
+              // This doesn't affect the test but the 'if' should be removed
+              // once the Chrome bug is fixed.
+              if (!shouldError) {
+                texture.destroy();
+              }
+            },
+            shouldError,
+            `size: ${size}, actualLimit: ${actualLimit}`
+          );
+        }
+      }
+    );
+  });
+
+g.test('getCurrentTexture,at_over')
+  .desc(`Test using at and over ${limit} limit`)
+  .params(kLimitBaseParams.combine('canvasType', kAllCanvasTypes))
+  .fn(async t => {
+    const { limitTest, testValueName, canvasType } = t.params;
+    await t.testDeviceWithRequestedLimits(
+      limitTest,
+      testValueName,
+      async ({ device, shouldError, testValue, actualLimit }) => {
+        for (let dimensionIndex = 0; dimensionIndex < 2; ++dimensionIndex) {
+          const size = [1, 1];
+          size[dimensionIndex] = testValue;
+
+          // This should not fail, even if the size is too large but it might fail
+          // if we're in a worker and HTMLCanvasElement does not exist.
+          const canvas = createCanvas(t, canvasType, size[0], size[1])!;
+          if (canvas) {
+            const context = canvas.getContext('webgpu') as GPUCanvasContext;
+            t.expect(!!context, 'should not fail to create context even if size is too large');
+
+            context.configure({
+              device,
+              format: getGPU().getPreferredCanvasFormat(),
+            });
```

Comment:
The spec says there should also be a WebGPU error from configure(), since the size is already too large before it's called:
https://gpuweb.github.io/gpuweb/#dom-gpucanvascontext-configure

Reply:
It's also useful to test a case where the size is _not_ too big, then configure is called (without generating an error), then the canvas is made too big, then getCurrentTexture is called.

I don't recall for sure but there's a chance we have a test for this already somewhere.

Reply:
If that wasn't clear, getCurrentTexture is tested in the getCurrentTexture,at_over test by first setting the size small so that configure succeeds, then setting it larger and calling getCurrentTexture

Reply:
> I think this comment came before my last push?

It did, yes, sorry.